src/psyclone/domain/common/transformations/kernel_module_inline_trans.py

```diff
@@ -207,22 +215,33 @@ def _prepare_code_to_inline(code_to_inline):
             if symbol.name not in code_to_inline.symbol_table:
                 code_to_inline.symbol_table.add(symbol)
             # And when necessary the modules where they come from
-            if symbol.is_import:
+            if symbol.is_unresolved:
+                # We don't know where this comes from, we need to bring
+                # in all top-level imports with wildcard imports
+                for mod in source_container.symbol_table.containersymbols:
+                    if mod.wildcard_import:
+                        if mod.name not in code_to_inline.symbol_table:
+                            code_to_inline.symbol_table.add(mod)
+                        else:
+                            code_to_inline.symbol_table.lookup(mod.name).\
+                                wildcard_import = True
+            elif symbol.is_import:
                 module_symbol = symbol.interface.container_symbol
                 if module_symbol.name not in code_to_inline.symbol_table:
                     code_to_inline.symbol_table.add(module_symbol)
-            elif symbol.is_unresolved:
-                for mod in source_container.symbol_table.containersymbols:
-                    if mod.name not in code_to_inline.symbol_table:
-                        code_to_inline.symbol_table.add(mod)
+                else:
+                    # If it already exists, we know its a container (from the
+                    # validation) so we just need to point to it
+                    symbol.interface._container_symbol = \
```

Comment:
Perhaps we should have a setter for `container_symbol`? If you agree please create a new issue and, for the time being, disable the pylint warning about accessing a protected member.

Reply:
Since it is just a straightforward setter property I implemented it in this PR instead of creating an issue.